### PR TITLE
[MDS-368] Snackbar v2

### PR DESCRIPTION
**Pull request by tkullisaar**

### CODE REVIEW

File: assets/ds-moon-preset.js [hunk 1]
```diff
@@ -123,6 +123,11 @@ module.exports = {
       },
       animation: {
         loader: "loader 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite",
+        rightslide: "rightslide 0.2s ease-in-out",
```
Comment (margusp2heathmont) [inline assets/ds-moon-preset.js:126]: Is this present defined in Moon React also? Can it be imported from there?

File: assets/js/hooks/snackbar.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const queue = [];
+
+const isVisible = (el) => !el.classList.contains("hidden") && !el.classList.contains("animate-fadeout")
+
+const visibleEls = () => Array.from(document.getElementsByClassName('moon-snackbar')).filter(isVisible)
```
Comment (Kypsis) [inline assets/js/hooks/snackbar.js:5]: Please don't use abbreviations for function names unless they are well defined industry standards. `El` and `Els` meaning here is maybe not ambiguous but it is still a good practice to be explicit, basically "don't-make-me-guess" principle.

File: assets/js/hooks/snackbar.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    el.dataset.to_hide = setTimeout(() => hideEl(el), el.dataset.timeout || 5000);
+}
+
+const showOrQueue = (el) => {
+    (el.dataset.in_queue !== undefined && !visibleEls().length) && queue.push(el) || showEl(el);
+}
+
+export default {
+    mounted() {
+        // console.log("mounted", this.el);
```
Comment (Kypsis) [inline assets/js/hooks/snackbar.js:27]: Leaving in commented code is generally considered a code smell. Especially if it is something as trivial as log statements.

File: lib/moon/design/snackbar.ex [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  def handle_event("toggle_open", _, socket) do
+    {:noreply, assign(socket, is_open: !socket.assigns.is_open)}
+  end
+
+  def on_click(id, _time) do
+    JS.add_class("hidden", to: ".moon-snackbar")
+    |> JS.remove_class("hidden", to: "##{id}")
+
+    # |> JS.add_class("hidden", to: "##{id}", time: time, transition: "animate-fadeout")
```
Comment (Kypsis) [inline lib/moon/design/snackbar.ex:40]: Leaving in commented code is generally considered a code smell.

File: lib/moon/design/snackbar/trigger.ex [hunk 5]
```diff
@@ -0,0 +1,18 @@
+defmodule Moon.Design.Snackbar.Trigger do
+  @moduledoc false
+
+  use Moon.StatelessComponent, slot: "trigger"
+
+  # slot(default, required: true)
```
Comment (Kypsis) [inline lib/moon/design/snackbar/trigger.ex:6]: Leaving in commented code is generally considered a code smell. And this seems like an extreme case, the whole body commented out?

File: lib/moon_web/examples/design/snackbar_example/auto_close.ex [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  use Moon.StatelessComponent
+
+  use MoonWeb, :example
+
+  alias Moon.Design.Snackbar
+  alias Moon.Design.Button
+
+  def render(assigns) do
+    ~F"""
+    <Snackbar id="snackbar_16" timeout={1000}>
```
Comment (Kypsis) [inline lib/moon_web/examples/design/snackbar_example/auto_close.ex:13]: Nit: better to extract those repeated timeout values out to common variables and avoid using magic numbers.

File: assets/ds-moon-preset.js (hunk 1 above)
Comment (phgrey) [reply assets/ds-moon-preset.js:126]: yeah, it's defined there the same way. we do not copy the ds-moon-preset.js for now, bc of few errors got when tried to

Comment (Kypsis) [reply assets/ds-moon-preset.js:126]: Also coupling together two repositories does not seem like a good idea, especially if they are derivative siblings of a source (the MDS Figma).

File: lib/moon_web/examples/design/snackbar_example/auto_close.ex (hunk 6 above)
Comment (phgrey) [reply lib/moon_web/examples/design/snackbar_example/auto_close.ex:13]: it's not a magic number, it's just a milliseconds in example. shown directly for transparency

**Review by phgrey — APPROVED** [review]

**Pull request merged**